h/static/scripts/controllers/autosuggest-dropdown-controller.js

```diff
@@ -0,0 +1,454 @@
+   *    element.
+   * @param {ConfigOptions} configOptions are used to set the initial set of items and the header
+   *    as well as providing the hooks for updates and callbacks
+   *
+   */
+  constructor(inputElement, configOptions) {
+
+    super(inputElement, configOptions);
+
+    if(!configOptions.renderListItem){
+      throw new Error('Missing renderListItem callback in AutosuggestDropdownController constructor');
+    }
+
+    if(!configOptions.listReducer){
+      throw new Error('Missing listReducer function in AutosuggestDropdownController constructor');
+    }
+
+    if(!configOptions.onSelect){
+      throw new Error('Missing onSelect callback in AutosuggestDropdownController constructor');
+    }
+
+    // set up our element class attribute enum values
+    // Note, we currently are not doing anything with the default
+    // classes, but we have them if we wanted to give something for a default
+    // styling.
+    if(configOptions.classNames){
+      this.options.classNames.container = configOptions.classNames.container || 'autosuggest__container';
+      this.options.classNames.list = configOptions.classNames.list || 'autosuggest__list';
+      this.options.classNames.item = configOptions.classNames.item || 'autosuggest__list-item';
+      this.options.classNames.activeItem = configOptions.classNames.activeItem || 'autosuggest__list-item--active';
+      this.options.classNames.header = configOptions.classNames.header || 'autosuggest__header';
+    }
+
+    // renaming simply to make it more obvious what
+    // the element is in other contexts the control
+    this._input = this.element;
+
+    // initial state values
+    this.setState({
+      visible: false,
+      header: configOptions.header || '',
+
+      // working list that are displayed to use
+      list: [],
+
+      // rootList is the original set that the reducer
+      // will receive to determine what should be shown
+      rootList: [],
+    });
+
+    this._setList(configOptions.list);
+  }
+
+  update(newState, prevState){
+
+    // if our prev state is empty then
+    // we assume that this is the first update/render call
+    if(!('visible' in prevState)){
+
+      // create the elements that make up the component
+      this._renderContentContainers();
+      this._addTopLevelEventListeners();
+    }
+
+    if(newState.visible !== prevState.visible){
+      // updates the dom to change the class which actually updates visibilty
+      setElementState(this._suggestionContainer, { open: newState.visible});
+    }
+
+    if(newState.header !== prevState.header){
+      this._header.innerHTML = newState.header;
+    }
+
+    if(this._listDidChange(newState.list, prevState.list)){
+      this._renderListItems();
+    }
+
+  }
+
+  /**
+   * _setHeader - sets what would be the top bolded header text
+   *  to give context of what the suggestions are for
+   *
+   * @param  {string} header Text to place in header
```

Comment:
Is this HTML or plain text? From the docs I would interpret it as plain text but looking at the code it treats this as HTML.